test/unit/models/testConfiguration/TestPreviewerConfigurationServiceTest.php

```diff
@@ -0,0 +1,112 @@
+<?php
+
+/**
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2020 (original work) Open Assessment Technologies SA
+ */
+
+declare(strict_types=1);
+
+namespace unit\models\testConfiguration;
+
+use oat\generis\test\TestCase;
+use oat\tao\model\providers\ProviderModule;
+use oat\taoQtiTest\models\runner\config\QtiRunnerConfig;
+use oat\taoQtiTestPreviewer\models\testConfiguration\service\TestPreviewerConfigurationService;
+use oat\taoQtiTestPreviewer\models\testConfiguration\TestPreviewerConfig;
+use oat\taoTests\models\runner\plugins\TestPlugin;
+use oat\taoTests\models\runner\plugins\TestPluginService;
+use oat\taoTests\models\runner\providers\TestProviderService;
+use PHPUnit\Framework\MockObject\MockObject;
+
+class TestPreviewerConfigurationServiceTest extends TestCase
+{
+
+    private const OPTIONS = ['c' => 'd'];
+
+    /**
+     * @var TestPreviewerConfigurationService
+     */
+    private $subject;
+    /**
+     * @var TestPluginService|MockObject
+     */
+    private $testPluginService;
+    /**
+     * @var QtiRunnerConfig|MockObject
+     */
+    private $qtiRunnerConfig;
+    /**
+     * @var TestProviderService|MockObject
+     */
+    private $testProviderService;
```

Comment:
```suggestion
    /**
     * @var TestPreviewerConfigurationService
     */
    private $subject;
    
    /**
     * @var TestPluginService|MockObject
     */
    private $testPluginService;
    
    /**
     * @var QtiRunnerConfig|MockObject
     */
    private $qtiRunnerConfig;
    
    /**
     * @var TestProviderService|MockObject
     */
    private $testProviderService;
```

```
empty lines :: https://www.php-fig.org/psr/psr-12/#23-lines

we should use only 1 line between methods 

new line between attributes

no line between consts

line and comment between blocks of consts 
```

https://oat-sa.atlassian.net/wiki/spaces/NEX/pages/311492833/CR+guidelines+for+Construct

Reply:
i would put before the required and then the subject attribute